src/event_listener.rs

```diff
@@ -172,19 +215,25 @@ impl<E: EntityEvent> On<E> {
         )
     }
 
-    /// Send an event `F`  any time this event listener is triggered.
-    pub fn send_event<F: Event + From<ListenerInput<E>>>() -> Self {
-        Self::run(
+    /// Send an event `F` any time this event listener is triggered.
+    pub fn send_event<F: Event + From<ListenerInput<E>>>(&mut self) -> ListenerHandle<E> {
+        self.run(
             move |event: Res<ListenerInput<E>>, mut ev: EventWriter<F>| {
                 ev.send(F::from(event.to_owned()));
             },
         )
     }
+    
+    /*
+
+    QUESTION: Do we still need this???
 
     /// Take the boxed system callback out of this listener, leaving an empty one behind.
     pub(crate) fn take(&mut self) -> CallbackSystem {
```

Comment:
This would be clearer if renamed to remove_all.

You can keep `take` around as a function, and add a deprecation warning.